File: javascript/034_matrix/matrix.js
export class Matrix {
  constructor(cadena_matrix) {
    this.matrix = cadena_matrix.split('\n');
  }

  get rows() {
    return this.matrix.map(row => row.split(' ').map(Number));
  }

  get columns() {
    return this.rows[0].map((_ , i) => this.rows.map(row => row[i]));
  }
}

File: python/084_matrix/matrix.py
class Matrix:
    def __init__(self, matrix_string):
        self.matrix = [i.split(' ') for i in matrix_string.split('\n')]

    def row(self, index):
        return [int(i) for i in self.matrix[index - 1]] 

    def column(self, index):
        return [int(i[index - 1]) for i in self.matrix]
